embedchain/apps/App.py

```diff
@@ -14,19 +16,27 @@ class App(EmbedChain):
     dry_run(query): test your prompt without consuming tokens.
     """
 
-    def __init__(self, config: AppConfig = None):
+    def __init__(self, config: AppConfig = None, system_prompt: Optional[str] = None):
         """
         :param config: AppConfig instance to load as configuration. Optional.
+        :param system_prompt: Optional. System prompt string.
         """
         if config is None:
             config = AppConfig()
 
```

Comment:
couldn't you just say

```python
if system_prompt and not config.system_prompt:
    config.system_prompt = system_prompt
```

right here, and use latch on to the existing implementation? do this for all app classes? should be easier to maintain. You should make a setter method in config though, to make it cleaner.

Reply:
So you are saying instead of using `self.system_prompt` we can set config.system_prompt for app classes. And add a setter method for `system_prompt` in config to set its value if we get prompt while app init. Right??

Reply:
I'm saying that instead of introducing something new we just use the App `__init__` to set the existing value in config, and keep working with that. As the most minimal way to achieve this.

My example should work, but introducing a `set_system_prompt` method in `BaseAppConfig` can be easier to understand.

Reply:
Ok but why are we using `BaseAppConfig` to do this when `QueryConfig` is used to pass prompt on chat/query level. It will take more changes and will same as current approach.

Reply:
Ok but why are we using `BaseAppConfig` when `QueryConfig` is used to get system_prompt on chat\query level. Using BaseAppConfig will take more changes and will be similar to current approach.

Reply:
I hope you aren't waiting for my input. I haven't had a chance to take a deeper look into this.

Reply:
Sorry I was away for few days but I don't think making changes in `BaseAppConfig` will be a short approach. So directly using  system_prompt from app `_init_` seems to me the shortest approach. Let me know if you find any better or else I'll make few minor changes for it and PR will be ready for review @cachho